pkg/pillar/vault/handler_zfs.go

```diff
@@ -375,7 +407,14 @@ func CreateZvolVault(log *base.LogObject, datasetName string, zfsKeyFile string,
 		return fmt.Errorf("Dataset %s available bytes read error: %v", parentDatasetName, err)
 	}
 
-	err = zfs.CreateVaultVolumeDataset(log, datasetName, zfsKeyFile, encrypted, sizeBytes)
+	if sizeBytes < zfs.ReserveEveStorageSizeGb {
+		//Bypass for dev/test VMs in eden
```

Comment:
Can't those tests set the CONFIG-PROPERTY for the percentage reservation so we can avoid test code in the product?

Reply:
Yes, I somehow missed that.  This etcd partition size should actually probably have its own config property.

Reply:
I was actually just talking with Pramodh about the boot option eve_install_kubevirt_reserve_for_eve_sizeGB, with the newer modified storage layout this value does seem redundant to the config-property for reservation storage.zfs.reserved.percent.  I have some cleanup I'll push up.

Reply:
Those two are different I think.
storage.zfs.reserved.percent is to make sure that ZFS performs well by leaving sine free/reserved space.
AFAIR the eve_install_kubevirt_reserve_for_eve_sizeGB is to designate how much EVE (logs, /persist/vault, /persist/clear, etc) need from /persist so we can give the rest to longhorn.

Making this more clear makes sense.
And if the EVE total and etcd space can be changed at runtime then they can be CONFIG-PROPERTIES as well.

Reply:
My latest commit removes the install option eve_install_kubevirt_reserve_for_eve_sizeGB and replaces that with eve_install_kubevirt_etcd_sizeGB.  The original motivation for eve_install_kubevirt_reserve_for_eve_sizeGB was due to the older storage layout allocating a very large volume for the kube container and we needed a little bit left over for eve.  The new storage layout instead allocates a small 10GB volume for k3s/etcd and all remaining space goes to vault/eve.